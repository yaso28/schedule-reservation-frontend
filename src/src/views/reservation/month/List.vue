<template>
  <div>
    <slot name="h1" />

    <CCard v-if="loaded">
      <CCardHeader>
        <CForm @submit.prevent="search">
          <CRow
            form
            class="form-group mb-0"
          >
            <CCol
              sm="6"
              class="form-inline"
            >
              <select
                id="search-year"
                v-model="searchYear"
                class="form-control"
              >
                <option
                  v-for="year in yearList"
                  :key="year"
                  :value="year"
                  :label="year"
                />
              </select>
              <label
                for="search-year"
                class="m-2"
              >
                {{ $t('month.year') }}
              </label>
              <select
                id="search-month"
                v-model="searchMonth"
                class="form-control"
              >
                <option
                  v-for="month in monthList"
                  :key="month"
                  :value="month"
                  :label="month"
                />
              </select>
              <label
                for="search-month"
                class="m-2"
              >
                {{ $t('month.month') }}{{ $t('month.later') }}
              </label>
            </CCol>
            <CCol sm="6">
              <CButton
                type="submit"
                color="info"
              >
                {{ $t('action.search') }}
              </CButton>
            </CCol>
          </CRow>
        </CForm>
      </CCardHeader>

      <CCardBody>
        <CDataTable
          :items="itemsFiltered"
          :fields="fields"
        >
          <template #reservation_status="{item}">
            <td>
              {{ item.reservation_status.name }}
            </td>
          </template>
          <template #schedule_status="{item}">
            <td>
              {{ item.schedule_status.name }}
            </td>
          </template>
          <template #action="{item}">
            <td>
              <CLink
                class="btn btn-outline-primary"
                :to="{ name: 'reservation.schedule.list', query: { from: item.first_day, to: item.last_day } }"
              >
                {{ $t('schedule.model') }}
              </CLink>
              <CLink
                class="btn btn-outline-primary"
                :to="{ name: 'reservation.reservation.list', query: { from: item.first_day, to: item.last_day } }"
              >
                {{ $t('reservation.model') }}
              </CLink>
              <CLink
                v-if="canWrite"
                class="btn btn-outline-primary"
                :to="{ name: 'reservation.month.send', params: { id: item.id } }"
              >
                {{ $t('send.model') }}
              </CLink>
            </td>
          </template>
        </CDataTable>
      </CCardBody>
    </CCard>
  </div>
</template>

<script>
import permission from '@/consts/permission'

export default {
  name: 'MonthList',
  inheritAttrs: false,
  inject: ['apiService', 'dateService'],
  data () {
    return {
      items: null,
      yearList: null,
      searchYear: null,
      searchMonth: null,
      currentYear: null,
      currentMonth: null,
      oldestYear: null,
      oldestMonth: null
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'reservation_status', label: this.$t('reservation_status.model') },
        { key: 'schedule_status', label: this.$t('schedule_status.model') },
        { key: 'action', label: '' }
      ]
    },
    monthList () {
      return Array.from(Array(12), (v, i) => i + 1)
    },
    loaded () {
      return this.items
    },
    itemsFiltered () {
      return (this.items ?? []).filter(
        item => this.currentYear < item.year || (this.currentYear === item.year && this.currentMonth <= item.month)
      )
    },
    canWrite () {
      return this.$store.getters['auth/can'][permission.reservation.write]
    }
  },
  mounted () {
    this.init()
  },
  methods: {
    init () {
      const today = this.dateService.today()
      this.searchYear = this.dateService.year(today)
      this.searchMonth = this.dateService.month(today)
      this.yearList = Array.from(Array(4), (v, i) => this.searchYear - 2 + i)
      this.search()
    },
    search () {
      this.currentYear = this.searchYear
      this.currentMonth = this.searchMonth
      if (!this.oldestYear || this.oldestYear > this.currentYear || (this.oldestYear === this.currentYear && this.oldestMonth > this.currentMonth)) {
        this.load()
      }
    },
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getMonthList(
          this.currentYear,
          this.currentMonth
        )
        this.oldestYear = this.currentYear
        this.oldestMonth = this.currentMonth
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
