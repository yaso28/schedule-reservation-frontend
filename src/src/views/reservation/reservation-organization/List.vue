<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #action="{item}">
            <td>
              <template v-if="canWrite">
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.reservation-organization.edit', params: { id: item.id } }"
                >
                  {{ $t('action.edit') }}
                </CLink>
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.reservation-organization.add', query: { original_id: item.id } }"
                >
                  {{ $t('action.copy') }}
                </CLink>
              </template>
            </td>
          </template>
        </CDataTable>
      </CCardBody>
      <CCardFooter v-if="canWrite">
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.reservation-organization.add'}"
        >
          {{ $t('action.add') }}
        </CLink>
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.reservation-organization.reorder'}"
        >
          {{ $t('action.reorder') }}
        </CLink>
      </CCardFooter>
    </CCard>
  </div>
</template>

<script>
import permission from '@/consts/permission'

export default {
  name: 'ReservationOrganizationList',
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'abbreviation', label: this.$t('common.abbreviation') },
        { key: 'registration_number', label: this.$t('reservation_organization.registration_number') },
        { key: 'action', label: '' }
      ]
    },
    canWrite () {
      return this.$store.getters['auth/can'][permission.reservation.write]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getReservationOrganizationList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
