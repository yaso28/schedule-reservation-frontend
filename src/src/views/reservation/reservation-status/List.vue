<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #reserved="{item}">
            <td>
              <MyBoolIcon :value="item.reserved" />
            </td>
          </template>
        </CDataTable>
      </CCardBody>
    </CCard>
  </div>
</template>

<script>
export default {
  name: 'ReservationStatusList',
  components: {
    MyBoolIcon: require('@/components/MyBoolIcon').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'description', label: this.$t('reservation_status.description') },
        { key: 'reserved', label: this.$t('reservation_status.reserved') }
      ]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getReservationStatusList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
