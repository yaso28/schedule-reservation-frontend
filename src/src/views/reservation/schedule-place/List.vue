<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #price_per_hour="{item}">
            <td class="text-right text-nowrap">
              {{ $n(item.price_per_hour, "currency") }}
            </td>
          </template>
          <template #action="{item}">
            <td>
              <template v-if="canWrite">
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-place.edit', params: { id: item.id } }"
                >
                  {{ $t('action.edit') }}
                </CLink>
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-place.add', query: { original_id: item.id } }"
                >
                  {{ $t('action.copy') }}
                </CLink>
              </template>
            </td>
          </template>
        </CDataTable>
      </CCardBody>
      <CCardFooter v-if="canWrite">
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-place.add'}"
        >
          {{ $t('action.add') }}
        </CLink>
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-place.reorder'}"
        >
          {{ $t('action.reorder') }}
        </CLink>
      </CCardFooter>
    </CCard>
  </div>
</template>

<script>
import permission from '@/consts/permission'

export default {
  name: 'SchedulePlaceList',
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'abbreviation', label: this.$t('common.abbreviation') },
        { key: 'price_per_hour', label: this.$t('schedule_place.price_per_hour') },
        { key: 'action', label: '' }
      ]
    },
    canWrite () {
      return this.$store.getters['auth/can'][permission.reservation.write]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getSchedulePlaceList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
