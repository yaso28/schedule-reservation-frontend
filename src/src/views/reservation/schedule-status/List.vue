<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #display_type="{item}">
            <td>
              <CBadge
                v-if="item.display_type"
                :color="item.display_type"
              >
                {{ item.display_type }}
              </CBadge>
            </td>
          </template>
          <template #is_public="{item}">
            <td>
              <MyBoolIcon :value="item.is_public" />
            </td>
          </template>
          <template #bulk_change_mode="{item}">
            <td>
              {{ $t(`schedule_status.bulk_change_mode.${item.bulk_change_mode}`) }}
            </td>
          </template>
        </CDataTable>
      </CCardBody>
    </CCard>
  </div>
</template>

<script>
export default {
  name: 'ScheduleStatusList',
  components: {
    MyBoolIcon: require('@/components/MyBoolIcon').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'display_type', label: this.$t('schedule_status.display_type') },
        { key: 'is_public', label: this.$t('schedule_status.is_public') },
        { key: 'bulk_change_mode', label: this.$t('schedule_status.bulk_change_mode.model') }
      ]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getScheduleStatusList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
