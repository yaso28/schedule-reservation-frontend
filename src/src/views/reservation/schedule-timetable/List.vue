<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #action="{item}">
            <td>
              <template v-if="canWrite">
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-timetable.edit', params: { id: item.id } }"
                >
                  {{ $t('action.edit') }}
                </CLink>
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-timetable.add', query: { original_id: item.id } }"
                >
                  {{ $t('action.copy') }}
                </CLink>
              </template>
            </td>
          </template>
        </CDataTable>
      </CCardBody>
      <CCardFooter v-if="canWrite">
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-timetable.add'}"
        >
          {{ $t('action.add') }}
        </CLink>
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-timetable.reorder'}"
        >
          {{ $t('action.reorder') }}
        </CLink>
      </CCardFooter>
    </CCard>
  </div>
</template>

<script>
import permission from '@/consts/permission'

export default {
  name: 'ScheduleTimetableList',
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'details', label: this.$t('schedule_timetable.details'), _classes: 'nl2br' },
        { key: 'action', label: '' }
      ]
    },
    canWrite () {
      return this.$store.getters['auth/can'][permission.reservation.write]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getScheduleTimetableList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
