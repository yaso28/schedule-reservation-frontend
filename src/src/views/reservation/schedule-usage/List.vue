<template>
  <div>
    <slot name="h1" />

    <CCard>
      <CCardBody>
        <CDataTable
          :items="items"
          :fields="fields"
        >
          <template #is_public="{item}">
            <td>
              <MyBoolIcon :value="item.is_public" />
            </td>
          </template>
          <template #reservation_organization="{item}">
            <td>
              {{ item.reservation_organization.name }}
            </td>
          </template>
          <template #action="{item}">
            <td>
              <template v-if="canWrite">
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-usage.edit', params: { id: item.id } }"
                >
                  {{ $t('action.edit') }}
                </CLink>
                <CLink
                  class="btn btn-outline-primary"
                  :to="{ name: 'reservation.schedule-usage.add', query: { original_id: item.id } }"
                >
                  {{ $t('action.copy') }}
                </CLink>
              </template>
            </td>
          </template>
        </CDataTable>
      </CCardBody>
      <CCardFooter v-if="canWrite">
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-usage.add'}"
        >
          {{ $t('action.add') }}
        </CLink>
        <CLink
          class="btn btn-outline-primary"
          :to="{ name: 'reservation.schedule-usage.reorder'}"
        >
          {{ $t('action.reorder') }}
        </CLink>
      </CCardFooter>
    </CCard>
  </div>
</template>

<script>
import permission from '@/consts/permission'

export default {
  name: 'ScheduleUsageList',
  components: {
    MyBoolIcon: require('@/components/MyBoolIcon').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      items: []
    }
  },
  computed: {
    fields () {
      return [
        { key: 'name', label: this.$t('common.name') },
        { key: 'is_public', label: this.$t('schedule_usage.is_public') },
        { key: 'reservation_organization', label: this.$t('reservation_organization.model') },
        { key: 'action', label: '' }
      ]
    },
    canWrite () {
      return this.$store.getters['auth/can'][permission.reservation.write]
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        this.items = await this.apiService.getScheduleUsageList()
      } catch (e) {
        if (!this.apiService.isHandledError(e)) {
          throw e
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    }
  }
}
</script>
