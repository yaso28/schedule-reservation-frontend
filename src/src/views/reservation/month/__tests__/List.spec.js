import List from '../List'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import promiseMocker from '@/testUtils/promiseMocker'
import storeModules from '@/store/modules'
import permission from '@/consts/permission'

jest.mock('@/store/modules')

const loadResult = require('@/testUtils/dummyApiData/month/list.json')
const route = { name: 'reservation.month.list' }
const mockToday = { year: 2020, month: 11, day: 18 }
const olderYm = { year: 2020, month: 10 }
const newerYm = { year: 2021, month: 1 }

async function search (wrapper, year, month) {
  await wrapper.find('#search-year').setValue(year)
  await wrapper.find('#search-month').setValue(month)
  await wrapper.find('form').trigger('submit')
}
function getSearchResultWrapper (wrapper) {
  return wrapper.find('tbody')
}

describe('List', () => {
  let mountOption
  let apiService
  let dateService
  beforeEach(() => {
    promiseMocker.init()
    mountOption = mountOptionCreator.create(storeModules)
    apiService = mountOption.provide.apiService
    dateService = mountOption.provide.dateService
    dateService.today = jest.fn().mockReturnValue(new Date(`${mockToday.year}-${mockToday.month}-${mockToday.day}`))
  })

  afterEach(() => {
    promiseMocker.clear()
    jest.resetAllMocks()
  })

  describe('write権限：あり', () => {
    beforeEach(() => {
      storeModules.auth.getters.can.mockReturnValue({ [permission.reservation.write]: true })
    })

    describe('マウント', () => {
      let wrapper
      let loadError
      let loadSwitch
      beforeEach(async () => {
        loadError = new Error('load')
        loadSwitch = promiseMocker.createSwitch()
        apiService.getMonthList = promiseMocker.mock(loadResult, loadError, loadSwitch)
        wrapper = await pageTester.mount(List, route, mountOption)
      })

      test('titleタグ', () => {
        pageTester.assertTitleTag()
      })

      test('metaタグ', () => {
        pageTester.assertMetaTags()
      })

      test('Processingカバー表示開始', () => {
        expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
      })

      test('データ取得処理を呼出', () => {
        expect(apiService.getMonthList).toHaveBeenCalledWith(mockToday.year, mockToday.month)
      })

      test('Processingカバー表示中', () => {
        expect(storeModules.processing.mutations.endGet).not.toHaveBeenCalled()
      })

      describe('データ取得：エラー', () => {
        beforeEach(() => {
          promiseMocker.setSwitch(loadSwitch, false)
          apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
          promiseMocker.flushPendings()
        })

        test('エラー処理', () => {
          expect(apiService.isHandledError).toHaveBeenCalledWith(loadError)
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })
      })

      describe('データ取得：成功', () => {
        beforeEach(() => {
          promiseMocker.setSwitch(loadSwitch, true)
          promiseMocker.flushPendings()
        })

        test('ページ全体表示', () => {
          expect(wrapper.element).toMatchSnapshot()
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })

        describe('古い年月を指定して検索', () => {
          beforeEach(async () => {
            storeModules.processing.mutations.beginGet.mockReset()
            storeModules.processing.mutations.endGet.mockReset()
            apiService.getMonthList = promiseMocker.mock(loadResult, loadError, loadSwitch)
            await search(wrapper, olderYm.year, olderYm.month)
          })

          test('Processingカバー表示開始', () => {
            expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
          })

          test('データ取得処理を呼出', () => {
            expect(apiService.getMonthList).toHaveBeenCalledWith(olderYm.year, olderYm.month)
          })

          test('Processingカバー表示中', () => {
            expect(storeModules.processing.mutations.endGet).not.toHaveBeenCalled()
          })
        })

        describe('新しい年月を指定して検索', () => {
          beforeEach(async () => {
            storeModules.processing.mutations.beginGet.mockReset()
            storeModules.processing.mutations.endGet.mockReset()
            apiService.getMonthList = promiseMocker.mock(loadResult, loadError, loadSwitch)
            await search(wrapper, newerYm.year, newerYm.month)
          })

          test('Processingカバーは表示されない', () => {
            expect(storeModules.processing.mutations.beginGet).not.toHaveBeenCalled()
          })

          test('データ取得処理は呼び出さない', () => {
            expect(apiService.getMonthList).not.toHaveBeenCalled()
          })

          test('検索結果表示', () => {
            expect(getSearchResultWrapper(wrapper).element).toMatchSnapshot()
          })
        })
      })
    })
  })

  describe('write権限：なし', () => {
    let wrapper
    beforeEach(async () => {
      storeModules.auth.getters.can.mockReturnValue({})
      apiService.getMonthList = promiseMocker.mockResolved(loadResult)
      wrapper = await pageTester.mount(List, route, mountOption)
      promiseMocker.flushPendings()
    })

    test('検索結果表示', () => {
      expect(getSearchResultWrapper(wrapper).element).toMatchSnapshot()
    })
  })
})
