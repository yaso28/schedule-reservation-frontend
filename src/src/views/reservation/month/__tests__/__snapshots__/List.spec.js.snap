// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List write権限：あり マウント metaタグ 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="2"
    name="robots"
  />,
]
`;

exports[`List write権限：あり マウント titleタグ 1`] = `"やそスポーツクラブ | 月一覧"`;

exports[`List write権限：あり マウント データ取得：成功 ページ全体表示 1`] = `
<div>
  <h1>
    月一覧
  </h1>
   
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <form
        class=""
      >
        <div
          class="form-row form-group mb-0"
        >
          <div
            class="form-inline col-sm-6"
          >
            <select
              class="form-control"
              id="search-year"
            >
              <option
                label="2018"
                value="2018"
              />
              <option
                label="2019"
                value="2019"
              />
              <option
                label="2020"
                value="2020"
              />
              <option
                label="2021"
                value="2021"
              />
            </select>
             
            <label
              class="m-2"
              for="search-year"
            >
              
              年
            
            </label>
             
            <select
              class="form-control"
              id="search-month"
            >
              <option
                label="1"
                value="1"
              />
              <option
                label="2"
                value="2"
              />
              <option
                label="3"
                value="3"
              />
              <option
                label="4"
                value="4"
              />
              <option
                label="5"
                value="5"
              />
              <option
                label="6"
                value="6"
              />
              <option
                label="7"
                value="7"
              />
              <option
                label="8"
                value="8"
              />
              <option
                label="9"
                value="9"
              />
              <option
                label="10"
                value="10"
              />
              <option
                label="11"
                value="11"
              />
              <option
                label="12"
                value="12"
              />
            </select>
             
            <label
              class="m-2"
              for="search-month"
            >
              
              月以降
            
            </label>
          </div>
           
          <div
            class="col-sm-6"
          >
            <button
              class="btn btn-info"
              type="submit"
            >
              
              検索
            
            </button>
          </div>
        </div>
      </form>
    </header>
     
    <div
      class="card-body"
    >
      <div
        data-v-5f7284b9=""
      >
        <!---->
        <div
          class="position-relative table-responsive"
          data-v-5f7284b9=""
        >
          <table
            class="table"
            data-v-5f7284b9=""
          >
            <thead
              data-v-5f7284b9=""
            >
              <tr
                data-v-5f7284b9=""
              >
                <th
                  class=""
                  data-v-5f7284b9=""
                  style="vertical-align: middle; overflow: hidden;"
                >
                  <div
                    data-v-5f7284b9=""
                  >
                    名称
                  </div>
                  <!---->
                </th>
                <th
                  class=""
                  data-v-5f7284b9=""
                  style="vertical-align: middle; overflow: hidden;"
                >
                  <div
                    data-v-5f7284b9=""
                  >
                    予約ステータス
                  </div>
                  <!---->
                </th>
                <th
                  class=""
                  data-v-5f7284b9=""
                  style="vertical-align: middle; overflow: hidden;"
                >
                  <div
                    data-v-5f7284b9=""
                  >
                    公開ステータス
                  </div>
                  <!---->
                </th>
                <th
                  class=""
                  data-v-5f7284b9=""
                  style="vertical-align: middle; overflow: hidden;"
                >
                  <div
                    data-v-5f7284b9=""
                  >
                    
                  </div>
                  <!---->
                </th>
              </tr>
              <!---->
            </thead>
            <tbody
              class="position-relative"
              data-v-5f7284b9=""
            >
              <tr
                data-v-5f7284b9=""
              >
                <td
                  class=""
                  data-v-5f7284b9=""
                >
                   2020年11月 
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            料金支払済
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            確定
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/schedule/list?from=2020-11-01&to=2020-11-30"
                    target="_self"
                  >
                    
              練習予定
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/reservation/list?from=2020-11-01&to=2020-11-30"
                    target="_self"
                  >
                    
              会場予約
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/month/send/2"
                    target="_self"
                  >
                    
              メール送信
            
                  </a>
                </td>
              </tr>
              <!---->
              <tr
                data-v-5f7284b9=""
              >
                <td
                  class=""
                  data-v-5f7284b9=""
                >
                   2020年12月 
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            電話予約済
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            未定
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/schedule/list?from=2020-12-01&to=2020-12-31"
                    target="_self"
                  >
                    
              練習予定
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/reservation/list?from=2020-12-01&to=2020-12-31"
                    target="_self"
                  >
                    
              会場予約
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/month/send/3"
                    target="_self"
                  >
                    
              メール送信
            
                  </a>
                </td>
              </tr>
              <!---->
              <tr
                data-v-5f7284b9=""
              >
                <td
                  class=""
                  data-v-5f7284b9=""
                >
                   2021年01月 
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            未予約
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  
            未定
          
                </td>
                <td
                  data-v-5f7284b9=""
                >
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/schedule/list?from=2021-01-01&to=2021-01-31"
                    target="_self"
                  >
                    
              練習予定
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/reservation/list?from=2021-01-01&to=2021-01-31"
                    target="_self"
                  >
                    
              会場予約
            
                  </a>
                   
                  <a
                    class="btn btn-outline-primary"
                    data-v-5f7284b9=""
                    href="/reservation/month/send/4"
                    target="_self"
                  >
                    
              メール送信
            
                  </a>
                </td>
              </tr>
              <!---->
              <!---->
            </tbody>
            <!---->
          </table>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
</div>
`;

exports[`List write権限：あり マウント データ取得：成功 新しい年月を指定して検索 検索結果表示 1`] = `
<tbody
  class="position-relative"
  data-v-5f7284b9=""
>
  <tr
    data-v-5f7284b9=""
  >
    <td
      class=""
      data-v-5f7284b9=""
    >
       2021年01月 
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            未予約
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            未定
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/schedule/list?from=2021-01-01&to=2021-01-31"
        target="_self"
      >
        
              練習予定
            
      </a>
       
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/reservation/list?from=2021-01-01&to=2021-01-31"
        target="_self"
      >
        
              会場予約
            
      </a>
       
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/month/send/4"
        target="_self"
      >
        
              メール送信
            
      </a>
    </td>
  </tr>
  <!---->
  <!---->
</tbody>
`;

exports[`List write権限：なし 検索結果表示 1`] = `
<tbody
  class="position-relative"
  data-v-5f7284b9=""
>
  <tr
    data-v-5f7284b9=""
  >
    <td
      class=""
      data-v-5f7284b9=""
    >
       2020年11月 
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            料金支払済
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            確定
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/schedule/list?from=2020-11-01&to=2020-11-30"
        target="_self"
      >
        
              練習予定
            
      </a>
       
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/reservation/list?from=2020-11-01&to=2020-11-30"
        target="_self"
      >
        
              会場予約
            
      </a>
       
      <!---->
    </td>
  </tr>
  <!---->
  <tr
    data-v-5f7284b9=""
  >
    <td
      class=""
      data-v-5f7284b9=""
    >
       2020年12月 
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            電話予約済
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            未定
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/schedule/list?from=2020-12-01&to=2020-12-31"
        target="_self"
      >
        
              練習予定
            
      </a>
       
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/reservation/list?from=2020-12-01&to=2020-12-31"
        target="_self"
      >
        
              会場予約
            
      </a>
       
      <!---->
    </td>
  </tr>
  <!---->
  <tr
    data-v-5f7284b9=""
  >
    <td
      class=""
      data-v-5f7284b9=""
    >
       2021年01月 
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            未予約
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      
            未定
          
    </td>
    <td
      data-v-5f7284b9=""
    >
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/schedule/list?from=2021-01-01&to=2021-01-31"
        target="_self"
      >
        
              練習予定
            
      </a>
       
      <a
        class="btn btn-outline-primary"
        data-v-5f7284b9=""
        href="/reservation/reservation/list?from=2021-01-01&to=2021-01-31"
        target="_self"
      >
        
              会場予約
            
      </a>
       
      <!---->
    </td>
  </tr>
  <!---->
  <!---->
</tbody>
`;
